File: backend/AgPolicy.Api/Models/Farmer.cs
namespace AgPolicy.Api.Models;

public class Farmer
{
    public int Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string County { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public List<Farm> Farms { get; set; } = [];
    public List<Quote> Quotes { get; set; } = [];
    public List<CropPolicy> Policies { get; set; } = [];
}

File: frontend/agpolicy-client/src/types/farmer.ts
export interface Farmer {
  id: number
  firstName: string
  lastName: string
  email: string
  phone?: string | null
  county: string
  state: string
}

export interface CreateFarmerRequest {
  firstName: string
  lastName: string
  email: string
  phone?: string
  county: string
  state: string
}

export interface Farm {
  id: number
  farmerId: number
  farmName: string
  acres: number
  county: string
  state: string
}

export interface CreateFarmRequest {
  farmName: string
  acres: number
  county: string
  state: string
}
